src/ts/greedy-nav/GreedyNav.ts

```diff
@@ -0,0 +1,1062 @@
+
+    return element.clientWidth - padding;
+};
+
+/**
+ * Get viewport size
+ * @returns {{width: number, height: number}}
+ */
+const viewportSize = () => {
+    const doc = document;
+    const w = window;
+    const docEl =
+        doc.compatMode && doc.compatMode === 'CSS1Compat'
+            ? doc.documentElement
+            : doc.body;
+
+    let width = docEl.clientWidth;
+    let height = docEl.clientHeight;
+
+    // mobile zoomed in?
+    if (w.innerWidth && width > w.innerWidth) {
+        width = w.innerWidth;
+        height = w.innerHeight;
+    }
+
+    return { width, height };
+};
+
+/**
+ * Count width of children and return the value
+ * @param e
+ */
+const getChildrenWidth = (e: HTMLElement) => {
+    const children = e.childNodes;
+    let sum = 0;
+    for (let i = 0; i < children.length; i++) {
+        if (children[i].nodeType !== 3) {
+            if (!Number.isNaN((<HTMLElement>children[i]).offsetWidth)) {
+                sum += (<HTMLElement>children[i]).offsetWidth;
+            }
+        }
+    }
+    return sum;
+};
+
+/**
+ * Get width
+ * @param elem
+ * @returns {number}
+ */
+const calculateWidths = (_this: HTMLElement, offsetPixels: number) => {
+    const totalWidth = getElementContentWidth(_this);
+    // Check if parent is the navwrapper before calculating its width
+
+    const restWidth = getChildrenWidth(_this) + offsetPixels;
+    const viewportWidth = viewportSize().width;
+
+    return { totalWidth, restWidth, viewportWidth };
+};
+
+export class GreedyNavMenu {
+    settings: Config;
+
+    count: number;
+
+    breaks: number[];
+
+    instance: number;
+
+    mainNavWrapper: Nullable<HTMLElement>;
+
+    navDropdown: Nullable<HTMLUListElement>;
+
+    navDropdownToggle: Nullable<HTMLElement>;
+
+    navDropdownToggleLabel: Nullable<HTMLSpanElement>;
+
+    toggleWrapper: Nullable<HTMLSpanElement>;
+
+    navDropdownSelector: string;
+
+    navDropdownToggleSelector: string;
+
+    mainNavSelector: string;
+
+    totalWidth: number;
+
+    restWidth: number;
+
+    viewportWidth: number;
+
+    constructor(config: Config = defaultConfig) {
+        this.settings = { ...defaultConfig, ...config };
+        this.count = 0;
+        this.breaks = [];
+        this.instance = 0;
+        this.mainNavWrapper = null;
+
+        this.navDropdown = null;
+        this.navDropdownToggle = null;
+        this.navDropdownToggleLabel = null;
+        this.toggleWrapper = null;
+
+        this.navDropdownSelector = `.${this.settings.navDropdownClassName}`;
+        this.navDropdownToggleSelector = `.${this.settings.navDropdownToggleClassName}`;
+        this.mainNavSelector = this.settings.mainNav;
+
+        this.totalWidth = 0;
+        this.restWidth = 0;
+        this.viewportWidth = 0;
+    }
+
+    init(): void {
+        // Feature test.
+        if (!supports && typeof Node === 'undefined') {
+            console.warn("This browser doesn't support priorityNav");
```

Comment:
Mixed branding here. `priorityNav` or `GreedyNav`?

Reply:
Agreed - this and the classes should be fixed.